pvlib/iotools/midc.py

```diff
@@ -236,4 +247,14 @@ def read_midc_raw_data_from_nrel(site, start, end, variable_map={}):
             'end': end.strftime('%Y%m%d')}
     endpoint = 'https://midcdmz.nrel.gov/apps/data_api.pl?'
     url = endpoint + '&'.join(['{}={}'.format(k, v) for k, v in args.items()])
-    return read_midc(url, variable_map=variable_map, raw_data=True)
+    # number of header columns and data columns do not always match,
+    # so first parse the header to determine the number of data columns
+    # to parse
+    csv_request = requests.get(url)
```

Comment:
Include `timeout` here? We did for `get_psm3` but I pushed for that addition and got the impression that `timeout` wasn't seen as necessary by all.

Reply:
Yes, something I overlooked and definitely a good idea to include.